frontend/public/components/dashboards-page/overview-dashboard/details-card.tsx

```diff
@@ -0,0 +1,140 @@
+import * as React from 'react';
+import * as _ from 'lodash-es';
+import { connect } from 'react-redux';
+
+import {
+  DashboardCard,
+  DashboardCardBody,
+  DashboardCardHeader,
+  DashboardCardTitle,
+} from '../../dashboard/dashboard-card';
+import { DetailsBody, DetailItem } from '../../dashboard/details-card';
+import { withDashboardResources, DashboardItemProps, FirehoseResource } from '../with-dashboard-resources';
+import { InfrastructureModel } from '../../../models';
+import { referenceForModel } from '../../../module/k8s';
+import { FLAGS } from '../../../const';
+import { featureReducerName, flagPending } from '../../../reducers/features';
+import { RootState } from '../../../redux';
+
+const OPENSHIFT_VERSION_QUERY = 'openshift_build_info{job="apiserver"}';
+
+const getClusterName = (infrastructure: object): string => {
+  const apiServerURL = _.get(infrastructure, 'status.apiServerURL');
+  if (!apiServerURL) {
+    return null;
+  }
+  const clusterName = apiServerURL.replace('https://api.', '');
+  const portIndex = clusterName.indexOf(':');
+  if (portIndex !== -1) {
+    return clusterName.slice(0, portIndex);
+  }
+  return clusterName;
+};
+
+const getInfrastructurePlatform = (infrastructure: object): string => _.get(infrastructure, 'status.platform');
+
+const getOpenshiftVersion = (openshiftClusterVersionResponse): string => {
+  const result = _.get(openshiftClusterVersionResponse, 'data.result', []);
+
+  // if cluster has more nodes, we take the version for the fist one
+  const firstCluster = Array.isArray(result) ? result[0] : result;
+  if (firstCluster) {
+    return _.get(firstCluster, 'metric.gitVersion');
+  }
+  return null;
+};
+
+const getKubernetesVersion = (kubernetesVersionResponse): string =>
```

Comment:
We have this logic in the about dialog. We should try to write it once and use it everywhere.

Reply:
Extracted the logic into separate functions in about dialog and they are now reused in Details Card. Not sure if there is a better place to move them to ?

Reply:
> Extracted the logic into separate functions in about dialog and they are now reused in Details Card. Not sure if there is a better place to move them to ?

I'd suggest `cluster-settings.ts`